### PR TITLE
Rewrite intersections

### PR DESCRIPTION
**Pull request by julialongtin**
rewrite the intersection code to use more restrictive types, and allow for examining the leader/follower when judging an intersection.

### CODE REVIEW

File: Graphics/Slicer/Machine/Infill.hs [hunk 1]
```diff
@@ -71,30 +71,26 @@ infillLineSegInside contour childContours line
...
+          getLineSegIntersections myline c = saneIntersections $ intersectsWith (Left myline) . Left <$> linesOfContour c
             where
               linesOfContour (PointSequence contourPoints) = makeLineSegsLooped $ (\(Point2 (x,y)) -> Point2 (roundToFifth x, roundToFifth y)) <$> contourPoints
-              saneIntersections :: [Intersection] -> [Point2]
-              saneIntersections xs = concat $ mapMaybe saneIntersection xs
+              saneIntersections :: [Either Intersection PIntersection] -> [Point2]
+              saneIntersections xs = catMaybes $ mapWithNeighbors saneIntersection xs
                 where
-                  saneIntersection :: Intersection -> Maybe [Point2]
+                  saneIntersection :: Either Intersection PIntersection -> Either Intersection PIntersection -> Either Intersection PIntersection -> Maybe Point2
```
Comment (isovector) [inline Graphics/Slicer/Machine/Infill.hs:80]: What do these arguments refer to? Why did this type change so dramatically? It feels smelly to have this many eithers in one type.

File: Graphics/Slicer/Machine/Infill.hs [hunk 2]
```diff
@@ -71,30 +71,26 @@ infillLineSegInside contour childContours line
...
+                  saneIntersection _                          (Left (HitStartPointL2 point)) (Left (HitEndPointL2   _)) = Just point
+                  saneIntersection (Left (HitStartPointL2 _)) (Left (HitEndPointL2   _    )) _                          = Nothing
+                  saneIntersection _                          (Left (HitEndPointL2   _    )) (Left (HitStartPointL2 _)) = Nothing
+                  saneIntersection (Left (HitEndPointL2   _)) (Left (HitStartPointL2 point)) _                          = Just point
+                  saneIntersection (Right PColinear)          (Left (HitStartPointL2 _    )) _                          = Nothing
+                  saneIntersection _                          (Left (HitEndPointL2   _    )) (Right PColinear)          = Nothing
+                  saneIntersection _                          (Right PColinear)              _                          = Nothing
+                  saneIntersection r1 r2 r3 = error $ "insane result of intersecting a line (" <> show myline <> ") with a contour " <> show c <> "\n" <> show r1 <> "\n" <> show r2 <> "\n" <> show r3 <> "\n"
```
Comment (isovector) [inline Graphics/Slicer/Machine/Infill.hs:92]: Avoid `error` calls. They only bring pain and sadness.

File: Graphics/Slicer/Machine/Infill.hs (hunk 1 above)
Comment (isovector) [reply Graphics/Slicer/Machine/Infill.hs:80]: This function is four `where` blocks deep. Consider moving it to a top level function. A good rule of thumb is "if you needed it here, you're probably going to need it somewhere else too." Aim to keep your function definitions <20 lines --- including their where blocks.

File: Graphics/Slicer/Math/Contour.hs [hunk 3]
```diff
@@ -151,13 +151,13 @@ contourContainsContour parent child = odd noIntersections
...
       | null contourPoints = []
-      | otherwise = mapMaybe (saneIntersection . lineIntersection line) $ makeLineSegsLooped contourPoints
-    saneIntersection :: Intersection -> Maybe Point2
-    saneIntersection (IntersectsAt p2) = Just p2
-    saneIntersection NoIntersection = Nothing
-    saneIntersection Parallel = Nothing
-    saneIntersection AntiParallel = Nothing
-    saneIntersection (Colinear _ _) = Nothing
+      | otherwise = mapMaybe (saneIntersection . intersectsWith (Left line) . Left) $ makeLineSegsLooped contourPoints
+    saneIntersection :: Either Intersection PIntersection -> Maybe Point2
```
Comment (isovector) [inline Graphics/Slicer/Math/Contour.hs:155]: Why did this type change? What's the meaning between left and right?

File: Graphics/Slicer/Math/Contour.hs [hunk 4]
```diff
@@ -219,17 +219,13 @@ intersections dstPoint contour srcPoint = saneIntersections l0 $ contourLineSegs
...
-            saneIntersection (IntersectsAt p2) = Just p2
-            saneIntersection NoIntersection = Nothing
-            saneIntersection Parallel = Nothing
-            saneIntersection AntiParallel = Nothing
-    -- FIXME: fix these cases. steal the code / algorithms from infillLineSegInside.
-    --          saneIntersection Collinear = Nothing
-    --          saneIntersection LColinear _ _ = Nothing
-    --          saneIntersection (HitStartPointL2 p2) = Just p2
-    --          saneIntersection (HitEndPointL2 p2) = Just p2
+            saneIntersection :: Either Intersection PIntersection -> Maybe Point2
```
Comment (isovector) [inline Graphics/Slicer/Math/Contour.hs:224]: Duplicate of `saneIntersection` above. Pull this out into a top level function.

File: Graphics/Slicer/Math/PGA.hs [hunk 5]
```diff
@@ -74,50 +67,126 @@ plinesIntersectIn pl1 pl2
...
 angleBetween pl1 pl2 =  scalarPart $ pv1 ⎣ pv2
   where
     (PLine2 pv1) = forcePLine2Basis (normalizePLine2 pl1)
     (PLine2 pv2) = forcePLine2Basis (normalizePLine2 pl2)
 
+-- | Translate a line a given amound along it's perpendicular bisector.
+translatePerp :: PLine2 -> ℝ -> PLine2
+translatePerp pl1 d = PLine2 $ addVecPair m $ rawPLine pl1
+  where
+    m = GVec [GVal (d*normOfPLine2 pl1) [GEZero 1]]
```
Comment (isovector) [inline Graphics/Slicer/Math/PGA.hs:101]: What is `m`? It could use a more descriptive name

File: Graphics/Slicer/Math/PGA.hs [hunk 6]
```diff
@@ -74,50 +67,126 @@ plinesIntersectIn pl1 pl2
...
 
--- | find the angle of intersection of two PLines.
+-- | Find the angle of intersection of two PLines.
 angleBetween :: PLine2 -> PLine2 -> ℝ
 angleBetween pl1 pl2 =  scalarPart $ pv1 ⎣ pv2
   where
     (PLine2 pv1) = forcePLine2Basis (normalizePLine2 pl1)
     (PLine2 pv2) = forcePLine2Basis (normalizePLine2 pl2)
 
+-- | Translate a line a given amound along it's perpendicular bisector.
```
Comment (isovector) [inline Graphics/Slicer/Math/PGA.hs:97]: typo

File: Graphics/Slicer/Math/PGA.hs [hunk 7]
```diff
@@ -74,50 +67,126 @@ plinesIntersectIn pl1 pl2
...
+translatePerp :: PLine2 -> ℝ -> PLine2
+translatePerp pl1 d = PLine2 $ addVecPair m $ rawPLine pl1
+  where
+    m = GVec [GVal (d*normOfPLine2 pl1) [GEZero 1]]
+    rawPLine (PLine2 a) = a
```
Comment (isovector) [inline Graphics/Slicer/Math/PGA.hs:102]: This should be a record accessor on `PLine2`

File: Graphics/Slicer/Math/PGA.hs [hunk 8]
```diff
@@ -74,50 +67,126 @@ plinesIntersectIn pl1 pl2
...
+
+-- | find the distance between a point and a line.
+distancePPointToPLine :: PPoint2 -> PLine2 -> ℝ
+distancePPointToPLine point line = normOfPLine2 $ join2PPoint2 point linePoint
+--error $ "result: " <> show (normOfPLine2 $ newLine) <> "\nLine: " <> show lvec <> "\nPoint: " <> show pvec <> "\nPerpLine: " <> show perpLine <> "\nLinePoint: " <> show linePoint <> "\nnewLine: " <> show newLine <> "\n"
+  where
+    (PLine2 lvec)  = normalizePLine2 line
+    (PPoint2 pvec) = canonicalizePPoint2 point
+    perpLine       = PLine2 $ lvec ⨅ pvec
```
Comment (isovector) [inline Graphics/Slicer/Math/PGA.hs:111]: The `PLine2` type doesn't seem like it's doing much work for you if you are packing *and* unpacking it in the same where block.

File: Graphics/Slicer/Math/PGA.hs [hunk 9]
```diff
@@ -74,50 +67,126 @@ plinesIntersectIn pl1 pl2
...
+
+-- | The Linear result of line segment intersection in 2 dimensions.
+data Intersection =
+    NoIntersection
+  | HitStartPointL2 Point2
+  | HitEndPointL2 Point2
+  deriving (Show)
+
+-- | an alias, for cases where either input is acceptable.
+type SegOrPLine2 = Either LineSeg PLine2
```
Comment (isovector) [inline Graphics/Slicer/Math/PGA.hs:127]: [Type aliases should only be used to concreteize types ](https://kowainik.github.io/posts/2019-02-06-style-guide#data-types). This isn't buying you anything, and just means readers have more things to keep paged.

File: Graphics/Slicer/Math/PGA.hs [hunk 10]
```diff
@@ -74,50 +67,126 @@ plinesIntersectIn pl1 pl2
...
+-- Check if/where lines/line segments intersect.
+-- entry point usable for all intersection needs, but it's faster to use the above two functions when you KNOW there MUST be an intersection.
+intersectsWith :: SegOrPLine2 -> SegOrPLine2 -> Either Intersection PIntersection
+intersectsWith (Left l1)   (Left l2)   =         lineIntersection    l1  l2
+intersectsWith (Right pl1) (Right pl2) = Right $ plinesIntersectIn   pl1 pl2
+intersectsWith (Left l1)   (Right pl1) =         lineIntersectsPLine l1  pl1
+intersectsWith (Right pl1) (Left l1)   =         lineIntersectsPLine l1  pl1
+
+-- Check if/where lines/line segments intersect.
+lineIntersectsPLine :: LineSeg -> PLine2 -> Either Intersection PIntersection
```
Comment (isovector) [inline Graphics/Slicer/Math/PGA.hs:138]: It seems to me like you want an actual `data` for this `Either` type, seeing as it keeps popping up everywhere.

File: Graphics/Slicer/Math/PGA.hs [hunk 11]
```diff
@@ -151,27 +220,22 @@ combineConsecutiveLineSegs lines
         sameLineSeg = meet2PLine2 (eToPLine2 l1) (eToPLine2 l2) == PPoint2 (GVec [])
         sameMiddlePoint = p2 == addPoints p1 s1
 
--- | Given the result of intersectionPoint, find out whether this intersection point is on the given segment, or not.
-onSegment :: LineSeg -> Point2 -> Bool
-onSegment (LineSeg p s) i =
-  sqNormOfPLine2 (join2PPoint2 (eToPPoint2 p) (eToPPoint2 i))               <= segmentLength &&
-  sqNormOfPLine2 (join2PPoint2 (eToPPoint2 i) (eToPPoint2 (addPoints p s))) <= segmentLength
+-- | find a point a given distance along a line perpendicularly bisecting this line at a given point.
+pointOnPerp :: LineSeg -> Point2 -> ℝ -> Point2
```
Comment (isovector) [inline Graphics/Slicer/Math/PGA.hs:224]: This type should be `LineSeg -> R -> Point2 -> Point2` so it forms an endomorphism and can be easily `fmap`ed and `modify`ed.

File: Graphics/Slicer/Math/PGA.hs [hunk 12]
```diff
@@ -151,27 +220,22 @@ combineConsecutiveLineSegs lines
...
+pointOnPerp line point d =
+--error $ "result: " <> show ( (motor•pvec)•reverse motor) <> "\nLine: " <> show lvec <> "\nPoint: " <> show pvec <> "\nPerpLine: " <> show perpLine <> "\nmotor•pvec: " <> show (motor•pvec) <> "\nmotor: " <> show motor <> "\nreverse motor: " <> show (reverse motor) <> "\n"
```
Comment (isovector) [inline Graphics/Slicer/Math/PGA.hs:226]: Might as well just delete this, instead of commenting it out. Same for error call above in this PR.

File: Graphics/Slicer/Math/PGA.hs [hunk 13]
```diff
@@ -235,12 +299,14 @@ ppointToPoint2 (PPoint2 (GVec vals)) = if infinitePoint
...
 
--- | Create a 2D projective line from a pair of euclidian endpoints.
+-- | Create a projective line from a euclidian line segment.
 eToPLine2 :: LineSeg -> PLine2
-eToPLine2 (LineSeg (Point2 (x1,y1)) (Point2 (x,y))) = PLine2 $ GVec $ foldl addVal [] [ GVal c [GEZero 1], GVal a [GEPlus 1], GVal b [GEPlus 2] ]
+eToPLine2 (LineSeg startPoint@(Point2 (x1,y1)) (Point2 (x,y))) = plineFromEndpoints startPoint (Point2 (x1+x,y1+y))
+
+-- | Create a projective line from a pair of euclidian points.
+plineFromEndpoints :: Point2 -> Point2 -> PLine2
+plineFromEndpoints (Point2 (x1,y1)) (Point2 (x2,y2)) = PLine2 $ GVec $ foldl addVal [] [ GVal c [GEZero 1], GVal a [GEPlus 1], GVal b [GEPlus 2] ]
```
Comment (isovector) [inline Graphics/Slicer/Math/PGA.hs:308]: **Always** use `foldl'` instead of `foldl`. The latter will always leak memory for absolutely no gain. It's a bug in the standard library.

File: Graphics/Slicer/Math/PGA.hs [hunk 14]
```diff
@@ -310,7 +376,7 @@ forcePLine2Basis (PLine2 pvec)
 -- | runtime basis coersion. ensure all of the '0' components exist on a PPoint2.
 forcePPoint2Basis :: PPoint2 -> PPoint2
 forcePPoint2Basis pt@(PPoint2 (GVec [GVal _ [GEZero 1, GEPlus 1], GVal _ [GEZero 1, GEPlus 2], GVal _ [GEPlus 1, GEPlus 2]])) = pt
-forcePPoint2Basis (PPoint2 pvec)                                                                                              =  PPoint2 $ forceBasis [[GEZero 1, GEPlus 1], [GEZero 1, GEPlus 2], [GEPlus 1, GEPlus 2]] pvec
+forcePPoint2Basis (PPoint2 pvec)                                                                                              = PPoint2 $ forceBasis [[GEZero 1, GEPlus 1], [GEZero 1, GEPlus 2], [GEPlus 1, GEPlus 2]] pvec
```
Comment (isovector) [inline Graphics/Slicer/Math/PGA.hs:379]: This line is too long.

**Review by isovector — CHANGES_REQUESTED** [review]

File: Graphics/Slicer/Machine/Infill.hs (hunk 1 above)
Comment (julialongtin) [reply Graphics/Slicer/Machine/Infill.hs:80]: Each either is a different intersection between a different line, and a line segment. I switched to using neighbor checking, so i could write intersection handlers that were neighbor aware.

File: Graphics/Slicer/Machine/Infill.hs (hunk 2 above)
Comment (julialongtin) [reply Graphics/Slicer/Machine/Infill.hs:92]: They should, in this case. i'm not done handling all of the intersections here.

File: Graphics/Slicer/Math/Contour.hs (hunk 3 above)
Comment (julialongtin) [reply Graphics/Slicer/Math/Contour.hs:155]: Intersections are linear-algebra intersections. PIntersections are Projective Geometric Algebra intersections.
the type changed by switching to intersectsWith, which now uses PIntersection types when possible, and Intersection types when necessary. Intersection types are only needed when dealing with line segments, instead of ideal lines.

File: Graphics/Slicer/Math/Contour.hs (hunk 4 above)
Comment (julialongtin) [reply Graphics/Slicer/Math/Contour.hs:224]: Probably not by the time i'm done with them. these should both expand to the "three at a time" form as in the other file, and need to handle colinear cases.

File: Graphics/Slicer/Machine/Infill.hs (hunk 2 above)
Comment (isovector) [reply Graphics/Slicer/Machine/Infill.hs:92]: A question I ask when reviewing PRs professionally --- "if you don't have time to do it correctly now, when will you?"

File: Graphics/Slicer/Math/Contour.hs (hunk 3 above)
Comment (isovector) [reply Graphics/Slicer/Math/Contour.hs:155]: Can `PIntersection` take a constructor which wraps `Intersection`? Without more context it's hard to say, but passing around lots and lots of eithers feels smelly as hell.

File: Graphics/Slicer/Math/Contour.hs (hunk 4 above)
Comment (isovector) [reply Graphics/Slicer/Math/Contour.hs:224]: I'd suggest to cross that bridge when you come to it, and to deduplicate in the meantime.

File: Graphics/Slicer/Math/PGA.hs (hunk 9 above)
Comment (julialongtin) [reply Graphics/Slicer/Math/PGA.hs:127]: This is trying to be an either or type.

File: Graphics/Slicer/Math/PGA.hs (hunk 10 above)
Comment (julialongtin) [reply Graphics/Slicer/Math/PGA.hs:138]: And this one is trying to be an (optional) union type.

File: Graphics/Slicer/Math/PGA.hs (hunk 13 above)
Comment (julialongtin) [reply Graphics/Slicer/Math/PGA.hs:308]: yeah, i knew this, just forgot it.

File: Graphics/Slicer/Math/PGA.hs [hunk 15]
```diff
@@ -135,12 +135,29 @@ intersectsWith (Right pl1) (Right pl2) = Right $ plinesIntersectIn   pl1 pl2
 intersectsWith (Left l1)   (Right pl1) =         lineIntersectsPLine l1  pl1
 intersectsWith (Right pl1) (Left l1)   =         lineIntersectsPLine l1  pl1
 
+-- | Check if/where two line segments intersect.
+lineIntersection :: LineSeg -> LineSeg -> Either Intersection PIntersection
+lineIntersection l1@(LineSeg p1 s1) l2@(LineSeg p2 s2)
+  | meet2PLine2 (eToPLine2 l1) (eToPLine2 l2) == PPoint2 (GVec [])         = Right PColinear
+  | onSegment l1 intersection && onSegment l2 intersection && intersection == p1 = Left $ HitStartPoint l1 intersection
```
Comment (isovector) [inline Graphics/Slicer/Math/PGA.hs:142]: I'd suggest pulling `onSegment l1 intersection && onsegment l2 intersection` into a where-binding, which would eliminate most of the duplicating here.

File: Graphics/Slicer/Machine/Infill.hs [hunk 16]
```diff
@@ -82,18 +80,19 @@ infillLineSegInside contour childContours line
                   saneIntersection _ (Left NoIntersection)         _ = Nothing
                   saneIntersection _ (Right PParallel)             _ = Nothing
                   saneIntersection _ (Right PAntiParallel)         _ = Nothing
-                  saneIntersection _                          (Left (HitStartPointL2 point)) (Left (HitEndPointL2   _)) = Just point
-                  saneIntersection (Left (HitStartPointL2 _)) (Left (HitEndPointL2   _    )) _                          = Nothing
-                  saneIntersection _                          (Left (HitEndPointL2   _    )) (Left (HitStartPointL2 _)) = Nothing
-                  saneIntersection (Left (HitEndPointL2   _)) (Left (HitStartPointL2 point)) _                          = Just point
-                  saneIntersection (Right PColinear)          (Left (HitStartPointL2 _    )) _                          = Nothing
-                  saneIntersection _                          (Left (HitEndPointL2   _    )) (Right PColinear)          = Nothing
+                  saneIntersection _                          (Left (HitStartPoint _ point)) (Left (HitEndPoint   _ _)) = Just point
```
Comment (isovector) [inline Graphics/Slicer/Machine/Infill.hs:83]: I don't understand why the cases in this function are what they are.  Why is `saneIntersection _ HitEndPoint{} HitStartPoint{}` nothing? Some comments or helper functions would really help aid in understanding of the logic here.

File: Graphics/Slicer/Machine/Infill.hs [hunk 17]
```diff
@@ -80,65 +80,68 @@ infillLineSegInside contour childContours line
...
-coveringLineSegsNegative :: Contour -> ℝ -> [LineSeg]
-coveringLineSegsNegative (PointSequence contourPoints) ls = flip LineSeg s . f <$> [-xMin,-xMin+lsX..xMax]
-    where s = Point2 (xMaxOutside,yMaxOutside)
-          f v = Point2 (v,0)
+coveringLineSegsNegative :: Contour -> ℝ -> [PLine2]
+coveringLineSegsNegative (PointSequence contourPoints) ls = eToPLine2 . flip LineSeg slope . f <$> [0,lss..(xMax-xMinRaw)+(yMax-yMin)+lss]
+    where slope = Point2 (1,1)
+          f v = Point2 (v-xDiff,0)
+          xDiff = (-(xMin-yMax))
+          yMin = minimum $ yOf <$> contourPoints
```
Comment (isovector) [inline Graphics/Slicer/Machine/Infill.hs:104]: `minimum` and `maximum` are partial functions that will explode if they're given given empty lists. Either change the definition to be `coveringLineSegsNegative (PointSquence contourPoints@(_:_))`, or change this to `maybe defaultValue getMinimum $ foldMap (Just . Min . yOf) contourPoints`

File: Graphics/Slicer/Machine/Infill.hs [hunk 18]
```diff
@@ -80,65 +80,68 @@ infillLineSegInside contour childContours line
...
 -- Generate lines over entire print area, where each one is aligned with a +1 slope.
--- FIXME: other ways to only generate covering lines over the given contour?
 -- FIXME: assumes we're in positive space.
-coveringLineSegsPositive :: Contour -> ℝ -> [LineSeg]
-coveringLineSegsPositive (PointSequence contourPoints) ls = flip LineSeg s . f <$> [0,lsY..yMax + xMax]
-    where s =  Point2 (xMaxOutside + yMaxOutside,- xMaxOutside - yMaxOutside)
-          f v = Point2 (0,v)
-          xMax = maximum $ xOf <$> contourPoints
+coveringLineSegsPositive :: Contour -> ℝ -> [PLine2]
+coveringLineSegsPositive (PointSequence contourPoints) ls = eToPLine2 . flip LineSeg slope . f <$> [0,lss..(xMax-xMin)+(yMax-yMin)+lss]
```
Comment (isovector) [inline Graphics/Slicer/Machine/Infill.hs:115]: I find the RHS hard to parse, it makes me need to think about the relative precedences of `.` and `<$>`. It would be much clearer if written with `fmap`.

File: Graphics/Slicer/Machine/Infill.hs [hunk 19]
```diff
@@ -80,65 +80,68 @@ infillLineSegInside contour childContours line
...
 
 -- Generate lines over entire print area, where each one is aligned with a -1 slope.
--- FIXME: other ways to only generate covering lines over the given contour?
 -- FIXME: assumes we're in positive space.
-coveringLineSegsNegative :: Contour -> ℝ -> [LineSeg]
-coveringLineSegsNegative (PointSequence contourPoints) ls = flip LineSeg s . f <$> [-xMin,-xMin+lsX..xMax]
-    where s = Point2 (xMaxOutside,yMaxOutside)
-          f v = Point2 (v,0)
+coveringLineSegsNegative :: Contour -> ℝ -> [PLine2]
+coveringLineSegsNegative (PointSequence contourPoints) ls = eToPLine2 . flip LineSeg slope . f <$> [0,lss..(xMax-xMinRaw)+(yMax-yMin)+lss]
```
Comment (isovector) [inline Graphics/Slicer/Machine/Infill.hs:100]: How is this function different than `coveringLineSegsNegative`? Can you rewrite the two of them as partially applied versions of some new `converingLineSegsHelper`? Otherwise you'll need to keep the two in sync forever, which is a great opportunity for bugs to pop up.

File: Graphics/Slicer/Machine/Infill.hs [hunk 20]
```diff
@@ -80,65 +80,68 @@ infillLineSegInside contour childContours line
...
-          yMax = maximum $ yOf <$> contourPoints
 
 -- Generate lines covering the entire contour, where each line is aligned with the X axis.
 -- FIXME: assumes we're in positive space.
-coveringLineSegsHorizontal :: Contour -> ℝ -> [LineSeg]
-coveringLineSegsHorizontal (PointSequence contourPoints) ls = flip LineSeg s . f <$> [yMin-ls,yMin..yMax]
-    where s =  Point2 (xMax+ls,0)
-          f v = Point2 (-ls,v)
+coveringLineSegsHorizontal :: Contour -> ℝ -> [PLine2]
+coveringLineSegsHorizontal (PointSequence contourPoints) ls = eToPLine2 . flip LineSeg slope . f <$> [yMin,yMin+ls..yMax]
```
Comment (isovector) [inline Graphics/Slicer/Machine/Infill.hs:140]: This and `coveringLineSegsVertical` could be unified into one function, and then partially applied to avoid the code duplication.

File: Graphics/Slicer/Math/PGA.hs (hunk 15 above)
Comment (julialongtin) [reply Graphics/Slicer/Math/PGA.hs]: good idea.

File: Graphics/Slicer/Machine/Infill.hs (hunk 16 above)
Comment (julialongtin) [reply Graphics/Slicer/Machine/Infill.hs:85]: added comments.

File: Graphics/Slicer/Machine/Infill.hs (hunk 17 above)
Comment (julialongtin) [reply Graphics/Slicer/Machine/Infill.hs:104]: Since this has a single entry point, i've added an emptyness change there.

File: Graphics/Slicer/Machine/Infill.hs (hunk 18 above)
Comment (julialongtin) [reply Graphics/Slicer/Machine/Infill.hs]: RHS?

File: Graphics/Slicer/Machine/Infill.hs (hunk 19 above)
Comment (julialongtin) [reply Graphics/Slicer/Machine/Infill.hs]: Tried to cover this a bit better with comments.

File: Graphics/Slicer/Machine/Infill.hs (hunk 20 above)
Comment (julialongtin) [reply Graphics/Slicer/Machine/Infill.hs:146]: Disagree. these four functions do the same thing, at different angles. eventually, there will be even more complicated functions for this stuff. these are the 'simple placeholders'.

File: Graphics/Slicer/Machine/Infill.hs (hunk 18 above)
Comment (isovector) [reply Graphics/Slicer/Machine/Infill.hs]: right hand side (of the equals sign)
